### ISSUE TITLE
Check your gradients with tf.compute_gradient_error - softmax applied twice.

### ISSUE THREAD

**Issue by pkubik** [vahidk/EffectiveTensorflow#9]
In the example code there is the `softmax` function activation just before `tf.nn.softmax_cross_entropy_with_logits`:
```
def non_differentiable_entropy(logits):
    probs = tf.nn.softmax(logits)
    return tf.nn.softmax_cross_entropy_with_logits(labels=probs, logits=logits)
```
The TensorFlow explicitly states that it is incorrect since it might lead to a vanishing gradient problem:
> WARNING: This op expects unscaled logits, since it performs a softmax on logits internally for efficiency. Do not call this op with the output of softmax, as it will produce incorrect results.

TL;DR - There should be no `softmax` before `tf.nn.softmax_cross_entropy_with_logits`.

**Comment by pkubik** [comment]
My bad seems like I didn't fully inspect the example.
